src/shared/extension/extension-settings-manager.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable boundaries/no-unknown-files */
+
+import { Hex } from 'shared/web3';
+
+import { ExtensionSettings } from './types';
+
+const extensionAddressBookSettingsStorageKeys = [
+  'idriss-send-enabled',
+  'wallet-lookup-enabled',
+] as const;
+
+export type ExtensionAddressBookSettingsStorageKeys =
+  (typeof extensionAddressBookSettingsStorageKeys)[number];
+
+const extensionGovernanceSettingsStorageKeys = [
+  'snapshot-enabled',
+  'tally-enabled',
+  'agora-enabled',
+] as const;
+
+export type ExtensionGovernanceSettingsStorageKeys =
+  (typeof extensionGovernanceSettingsStorageKeys)[number];
+
+const extensionIntegrationSettingsStorageKeys = [
+  'polymarket-enabled',
+  'gitcoin-enabled',
+] as const;
+
+export type ExtensionIntegrationSettingsStorageKeys =
+  (typeof extensionIntegrationSettingsStorageKeys)[number];
+
+export const extensionSettingsStorageKeys = [
+  'entire-extension-enabled',
+  ...extensionAddressBookSettingsStorageKeys,
+  ...extensionGovernanceSettingsStorageKeys,
+  ...extensionIntegrationSettingsStorageKeys,
+] as const;
+
+export type ExtensionSettingsStorageKey =
+  (typeof extensionSettingsStorageKeys)[number];
+
+interface StoredWallet {
+  account: Hex;
+  providerRdns: string;
+}
+
+export const ExtensionSettingsManager = {
+  setSettings(settings: Partial<ExtensionSettings>) {
+    chrome.storage.local.set(settings).catch(console.error);
+  },
+
+  saveWallet(wallet: StoredWallet) {
+    return chrome.storage.local.set({
+      'idriss-wallet': JSON.stringify(wallet),
+    });
+  },
+
+  clearWallet() {
+    return chrome.storage.local.remove('idriss-wallet');
+  },
+
+  getWallet(): Promise<StoredWallet | undefined> {
+    return new Promise((resolve) => {
+      void chrome.storage.local.get('idriss-wallet').then((storedWalletRaw) => {
+        const storedWallet = storedWalletRaw['idriss-wallet']
+          ? (JSON.parse(storedWalletRaw['idriss-wallet']) as StoredWallet)
+          : undefined;
+        return resolve(storedWallet);
+      });
+    });
+  },
+
+  getAllSettings(): Promise<Record<ExtensionSettingsStorageKey, boolean>> {
+    return new Promise((resolve) => {
+      void chrome.storage.local.get().then((result) => {
+        return resolve(result as Record<ExtensionSettingsStorageKey, boolean>);
+      });
+    });
```

Comment:
one thing which worries me is that this is now pointing to everything which is stored inside localStorage, settings should be hidden under some field name like `settings` so:
```ts
chrome.storage.local.get(SETTINGS_STORAGE_KEY);
```

also I believe that this type is a bit lieing, what if this is a fresh user and he had never saved anything in chrome storage? Probably we won't get whole dictionary of settings, what we should probably do is something like this:
```tsx
     void chrome.storage.local.get().then((result) => {
        const settingsFromStorage = result as Partial<Record<ExtensionSettingsStorageKey, boolean>>;
        const settings = { ...DEFAULT_SETTINGS, ...settingsFromStorage }; // not sure if type here will be without Partial, if not let's type cast
        return resolve(settings);
      });
```